File: src/main/java/ro/mpp2024/Domain/Participare.java
package ro.mpp2024.Domain;

import java.time.LocalDate;
import java.util.Objects;

public class Participare extends Entity<Pair<Integer,Integer>>{
    private LocalDate dataInscrierii;

    public Participare(LocalDate dataInscrierii) {
        this.dataInscrierii = dataInscrierii;
    }

    public LocalDate getDataInscrierii() {
        return dataInscrierii;
    }

    public void setDataInscrierii(LocalDate dataInscrierii) {
        this.dataInscrierii = dataInscrierii;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        Participare that = (Participare) o;
        return Objects.equals(dataInscrierii, that.dataInscrierii);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), dataInscrierii);
    }

    @Override
    public String toString() {
        return "ParticipareValdiator{" +
                getId().toString() +
                ", dataInscrierii=" + dataInscrierii +
                '}';
    }
}

File: SwimmingCompetitionCSharpGUI/Domain/Participare.cs
namespace SwimmingCompetition.Domain;

public class Participare:Entity<Pair<int, int>>
{
    public DateOnly DataInscrierii { get; }

    public Participare(DateOnly dataInscrierii)
    {
        DataInscrierii = dataInscrierii;
    }

    protected bool Equals(Participare other)
    {
        return base.Equals(other) && DataInscrierii.Equals(other.DataInscrierii);
    }

    public override bool Equals(object? obj)
    {
        if (ReferenceEquals(null, obj)) return false;
        if (ReferenceEquals(this, obj)) return true;
        if (obj.GetType() != this.GetType()) return false;
        return Equals((Participare)obj);
    }

    public override int GetHashCode()
    {
        return HashCode.Combine(base.GetHashCode(), DataInscrierii);
    }

    public override string ToString()
    {
        return $"{base.ToString()}, {nameof(DataInscrierii)}: {DataInscrierii}";
    }
}